clients/client/src/main/java/org/projectnessie/client/http/impl/jdk11/JavaRequest.java

```diff
@@ -0,0 +1,431 @@
+
+/** Implements Nessie HTTP request processing using Java's new {@link HttpClient}. */
+@SuppressWarnings("Since15") // IntelliJ warns about new APIs. 15 is misleading, it means 11
+final class JavaRequest extends BaseHttpRequest {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(JavaRequest.class);
+
+  private final HttpClient client;
+
+  JavaRequest(JavaHttpClient client) {
+    super(client.config);
+    this.client = client.client;
+  }
+
+  @Override
+  public org.projectnessie.client.http.HttpResponse executeRequest(Method method, Object body)
+      throws HttpClientException {
+
+    URI uri = uriBuilder.build();
+
+    HttpRequest.Builder request =
+        HttpRequest.newBuilder().uri(uri).timeout(Duration.ofMillis(config.getReadTimeoutMillis()));
+
+    RequestContext context = new RequestContextImpl(headers, uri, method, body);
+
+    boolean doesOutput = prepareRequest(context);
+
+    for (HttpHeader header : headers.allHeaders()) {
+      for (String value : header.getValues()) {
+        request = request.header(header.getName(), value);
+      }
+    }
+
+    BodyPublisher bodyPublisher =
+        doesOutput ? BodyPublishers.fromPublisher(new Submitter(context)) : BodyPublishers.noBody();
+    request = request.method(method.name(), bodyPublisher);
+
+    HttpResponse<InputStream> response = null;
+    try {
+      try {
+        LOGGER.debug("Sending {} request to {} ...", method, uri);
+        response = client.send(request.build(), BodyHandlers.ofInputStream());
+      } catch (HttpConnectTimeoutException e) {
+        throw new HttpClientException(
+            String.format(
+                "Timeout connecting to '%s' after %ds",
+                uri, config.getConnectionTimeoutMillis() / 1000),
+            e);
+      } catch (HttpTimeoutException e) {
+        throw new HttpClientReadTimeoutException(
+            String.format(
+                "Cannot finish %s request against '%s'. Timeout while waiting for response with a timeout of %ds",
+                method, uri, config.getReadTimeoutMillis() / 1000),
+            e);
+      } catch (MalformedURLException e) {
+        throw new HttpClientException(
+            String.format("Cannot perform %s request. Malformed Url for %s", method, uri), e);
+      } catch (IOException e) {
+        throw new HttpClientException(
+            String.format("Failed to execute %s request against '%s'.", method, uri), e);
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+
+      JavaResponseContext responseContext = new JavaResponseContext(response);
+
+      List<BiConsumer<ResponseContext, Exception>> callbacks = context.getResponseCallbacks();
+      if (callbacks != null) {
+        callbacks.forEach(callback -> callback.accept(responseContext, null));
+      }
+
+      config.getResponseFilters().forEach(responseFilter -> responseFilter.filter(responseContext));
+
+      if (response.statusCode() >= 400) {
```

Comment:
Should we have this logic in `UrlConnectionRequest` too?

Reply:
Duh - this code is actually superfluous, removed. (`NessieHttpResponseFilter`)